src/MOI_wrapper.jl

```diff
@@ -0,0 +1,229 @@
+"""
+Constructs a Differentiable Optimizer model from a MOI Optimizer.
+Supports `forward` and `backward` methods for solving and differentiating the model respectectively.
+
+## Note 
+Currently supports differentiating linear and quadratic programs only.
+"""
+
+const VI = MOI.VariableIndex
+const CI = MOI.ConstraintIndex
+
+
+mutable struct Optimizer <: MOI.AbstractOptimizer
+    optimizer::MOI.Bridges.LazyBridgeOptimizer
+    primal_optimal::Array{Float64}  # solution
+    dual_optimal::Array{Float64}  
+    var_idx::Vector{VI}
+    con_idx::Vector{CI}
+
+    function Optimizer(optimizer_constructor::Type)
```

Comment:
Just do `(optimizer_constructor)`, it may not be a `Type`, it could also be a `Function` or a `MOI.OptimizerWithAttributes`, ...

Reply:
done